### PR TITLE
bump test-infra to bazel 0.7.0

### PR DESCRIPTION
**Pull request by BenTheElder**
0.7.0 has been a smooth upgrade so far locally, let's try it on test-infra.

### CODE REVIEW

**Comment by BenTheElder** [comment]
/area bazel
/area jobs
cc @ixdy

**Comment by krzyzacy** [comment]
perfect merging time

/lgtm

**Comment by BenTheElder** [comment]
(watching this, should be a noop)

**Comment by BenTheElder** [comment]
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/pr-logs/pull/test-infra/5088/pull-test-infra-bazel/9044/

```
W1021 01:18:42.987] Run: ('bazel', 'clean', '--expunge')
W1021 01:18:44.258] INFO: $TEST_TMPDIR defined: output root default is '/root/.cache/bazel'.
W1021 01:18:44.262] Extracting Bazel installation...
W1021 01:19:27.267] .
W1021 01:19:27.267] unexpected pipe read status: (error: 22): Invalid argument
W1021 01:19:27.267] Server presumed dead. Now printing '/root/.cache/bazel/_bazel_root/27a95fe2fc68aa1195ddf1a4bbaaea62/server/jvm.out':
W1021 01:19:27.267] Traceback (most recent call last):
W1021 01:19:27.267]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_bazel.py", line 242, in <module>
W1021 01:19:27.268]     main(parse_args())
W1021 01:19:27.268]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_bazel.py", line 126, in main
W1021 01:19:27.268]     check('bazel', 'clean', '--expunge')
W1021 01:19:27.268]   File "/workspace/./test-infra/jenkins/../scenarios/kubernetes_bazel.py", line 33, in check
W1021 01:19:27.268]     subprocess.check_call(cmd)
W1021 01:19:27.268]   File "/usr/lib/python2.7/subprocess.py", line 541, in check_call
W1021 01:19:27.268]     raise CalledProcessError(retcode, cmd)
W1021 01:19:27.268] subprocess.CalledProcessError: Command '('bazel', 'clean', '--expunge')' returned non-zero exit status 37
E1021 01:19:27.272] Command failed
I1021 01:19:27.272] process 409 exited with code 1 after 0.8m
E1021 01:19:27.272] FAIL: pull-test-infra-bazel
```

**Comment by BenTheElder** [comment]
This happened twice now on that PR, very strange failure mode in CI. Reverting. Will reattempt this later :confused:

File: images/bazelbuild/Makefile [hunk 1]
```diff
@@ -15,7 +15,8 @@
 IMG = gcr.io/k8s-testimages/bazelbuild
 TAG = $(shell date +v%Y%m%d)-$(shell git describe --tags --always --dirty)
 
-BAZEL_VERSIONS=0.5.2 0.5.4 0.6.1
+# TODO(bentheelder): retire 0.6.1 in favor of 0.7.0
```
Comment (ixdy) [inline images/bazelbuild/Makefile]: we should be able to retire 0.5.4 too, I think

**Pull request merged**